### PR TITLE
Add Coingecko market data hook

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

File: packages/app/utils/coin-gecko/index.tsx [hunk 1]
```diff
@@ -22,3 +56,23 @@ export const useTokenPrice = <T extends string>(tokenId: T) => {
  * Fetch current Send token price
  */
 export const useSendPrice = () => useTokenPrice('send-token' as const)
+
+/**
+ * Fetch coin market data
+ */
+export const useTokenMarketData = <T extends coins[number]['coingeckoTokenId']>(tokenId: T) => {
```
Comment (0xBigBoss) [inline packages/app/utils/coin-gecko/index.tsx:63]: let's try increasing the stale time on the coin gecko hook something like 5minutes is completely acceptable https://tanstack.com/query/v5/docs/framework/react/guides/caching

that should help us avoid unnecessary network requests.

File: packages/app/utils/coin-gecko/index.tsx [hunk 2]
```diff
@@ -1,9 +1,43 @@
 import { useQuery } from '@tanstack/react-query'
+import type { coins } from 'app/data/coins'
+
+type MarketData<T extends coins[number]['coingeckoTokenId']> = {
```
Comment (0xBigBoss) [inline packages/app/utils/coin-gecko/index.tsx:4]: we should consider just using a zod type here and safe parse.

Comment (youngkidwarrior) [reply packages/app/utils/coin-gecko/index.tsx:4]: sure thing

**Review by 0xBigBoss — APPROVED** [review]

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Apr 22, 5:21 PM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/327).

**Pull request merged**